src/components/ContactDetails/ContactDetails.tsx

```diff
@@ -1,119 +1,81 @@
-    borderColor: theme.palette.primary.dark,
-  },
-  wrapper: {
+  loader: {
     display: 'flex',
-    flexDirection: 'row',
-    marginTop: theme.spacing(2),
-  },
-  avatar: {
-    width: '10rem',
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '100vh',
+    width: '100%',
+    marginTop: theme.spacing(-7),
   },
 }));
 
 const ContactDetails: React.FC<ContactDetailsProps> = props => {
   const { newbieId } = useParams<QueryNewbieArgs>();
-  const classes = useStyles();
-  const { loading, data } = useQuery<Query, QueryNewbieArgs>(CONTACT_DETAILS, {
-    variables: { newbieId },
-  });
-
-  const renderContactDetails = () => (
-    <div data-testid='contact-details-page'>
-      <Box className={classes.wrapper}>
-        <Box className={classes.avatar}>
-          <Avatar imgSrc={data && data.newbie.photo} />
-        </Box>
-        <Box>
-          <Typography component='p'>
-            <Box component='strong' fontWeight={'fontWeightBold'}>
-              Name
-            </Box>
-          </Typography>
-
-          <Typography component='p' data-testid='contact-name'>
-            {data && data.newbie.name}
-          </Typography>
-
-          <Typography component='p'>
-            <Box component='strong' fontWeight={'fontWeightBold'}>
-              What I do
-            </Box>
-          </Typography>
-          <Typography component='p'>{data && data.newbie.position}</Typography>
+  const { buddyId } = useParams<QueryBuddyArgs>();
+  const { data: AuthData } = useContext<AuthContextData>(AuthContext);
+  const { role } = AuthData;
 
-          <Typography component='p'>
-            <Box component='strong' fontWeight={'fontWeightBold'}>
-              Start date
-            </Box>
-          </Typography>
-          <Typography component='p'>{data && data.newbie.startDate}</Typography>
+  const handleBackClick = () => {
+    props.history.push(ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', newbieId));
+  };
 
-          <Typography component='p'>
-            <Box component='strong' fontWeight={'fontWeightBold'}>
-              E-mail
-            </Box>
-          </Typography>
-          <Typography component='p'>{data && data.newbie.email}</Typography>
-
-          <Typography component='p'>
-            <Box component='strong' fontWeight={'fontWeightBold'}>
-              Phone
-            </Box>
-          </Typography>
-          <Typography component='p'>{data && data.newbie.phoneNumber}</Typography>
+  const getQueryByRole = (role: UserRole) => {
+    if (isBuddy(role)) {
+      return {
+        query: NEWBIE_CONTACT_DETAILS,
+        variables: { newbieId },
+        userRole: UserRole.Newbie.toLowerCase(),
+      };
+    } else if (isNewbie(role)) {
+      return {
+        query: BUDDY_CONTACT_DETAILS,
+        variables: { buddyId },
+        userRole: UserRole.Buddy.toLowerCase(),
+      };
+    }
+  };
+  const { query, variables, userRole } = getQueryByRole(role) || {};
```

Comment:
As a suggestion - seems this part can be updated to use less code and resources:
```
const QUERY_BY_ROLE = {
  [UserRole.Buddy]: {
        query: BUDDY_CONTACT_DETAILS,
        variables: { buddyId },
        userRole: UserRole.Buddy.toLowerCase(),
      },
  [UserRole.Newbie]: {
        query: NEWBIE_CONTACT_DETAILS,
        variables: { newbieId },
        userRole: UserRole.Newbie.toLowerCase(),
      },
}; 
const { query, variables, userRole } = QUERY_BY_ROLE[role];
```

Reply:
Thanks @Tonik1204 for a tip :+1: . That's slightly cleaner approach. I've added it here: f7dd3b5188250d097bfebbf8063d64d015dce6b5